File: backend/credit-wizard-api/credit-wizard-api/Dtos/DegreeProgressDto.cs
namespace credit_wizard_api.Dtos;

public class DegreeProgressDto
{
    public Guid UserId { get; set; }
    public int TotalDegreeEtcsPoints { get; set; }
    public int ReachedEtcsPoints { get; set; }
    public int OpenEtcsPoints { get; set; }
    public int MissingEtcsPoints { get; set; }
    public int MissedEctsPoints { get; set; }
    public int PercentageFinishedRequiredModuls { get; set; }
}

File: frontend/credit-wizard/src/app/shared/dtos/degreeProgressDto.ts
import { Guid } from "guid-typescript";

export interface IDegreeProgressDto{
    userId: Guid,
    totalDegreeEtcsPoints: number,
    reachedEtcsPoints: number,
    openEtcsPoints: number,
    missingEtcsPoints: number,
    missedEctsPoints: number,
    percentageFinishedRequiredModuls: number
}